packages/react/src/store/sagas/backendSaga.ts

```diff
@@ -133,9 +135,32 @@ function* handlePing() {
   );
 }
 
+function* handleCheckInternetAccess() {
+  try {
+    const response:boolean = yield window.backend.checkInternetAccess();
+    if (response){
+      return;
+    }
+  } catch(err) {}
+  yield put(
+    setCurrentMessageUpdate([{
+      messageId: uuid(),
+      groupId: "internetAccessFailed",
+      type: "error",
+      header: "No Internet Access",
+      content: "Ensure you have internet access before using the application",
+      buttonText: "View prerequisites",
+      onButtonClick: () => window.electron.openExternalUrl(externalUrls.prereq)
+    }])
+  );
+}
+
 function* handleAnalyzeSolution(action: ReturnType<typeof analyzeSolution.request>) {
   yield put(ping());
+  yield put(checkInternetAccess());
   try {
+    // call backend method again because error will halt assessment
+    yield window.backend.checkInternetAccess();
```

Comment:
why we still add this check here ?

Reply:
removed as a part of change to util function